src/mqtt/mod.rs

```diff
@@ -218,6 +225,109 @@ async fn listen_on_camera(cam_config: Arc<CameraConfig>, mqtt_config: &MqttConfi
     Ok(())
 }
 
+/// Enables MQTT discovery for a camera. See docs at https://www.home-assistant.io/integrations/mqtt/#mqtt-discovery
+async fn enable_discovery(
```

Comment:
First time writing a real rust function, would appreciate pointers on anything that's not idiomatic!